sw/device/silicon_creator/lib/drivers/watchdog.c

```diff
@@ -5,42 +5,59 @@
 #include "sw/device/silicon_creator/lib/drivers/watchdog.h"
 
 #include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
 
 #include "aon_timer_regs.h"
 #include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
 #include "pwrmgr_regs.h"
 
 enum {
   kBase = TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR,
   kPwrMgrBase = TOP_EARLGREY_PWRMGR_AON_BASE_ADDR,
-  // The AON domain is always clocked at 200 Hz.
-  // TODO(lowRISC/opentitan#7385): update this after the AON frequency is
-  // formally defined in the project.
-  kAonTimerRate = 200000,
 };
 
-void watchdog_init(uint32_t timeout_ms) {
+void watchdog_init(lifecycle_state_t lc_state) {
+  // Determine the threshold cycles.
+  uint32_t threshold = 0;
+  switch (lc_state) {
+    case kLcStateTest:
+    case kLcStateRma:
+      // Set the watchdog bite threshold to a high default value when in test
+      // and RMA lifecycle states.
+      threshold = UINT32_MAX;
```

Comment:
I'm not sure about this, it might be better to disable the watchdog in these lifecycle states.

Reply:
yeah in test states definitely disable, especially RMA. 

Honestly even in production states I'm kind of iffy about enabling watchdog... this would imply that if you see a field failure it would be quite difficult to debug..

Reply:
Hmm, but as long as the threshold is set to a proper value and rom_ext and/or owner code disables it or handles it properly watchdog shouldn't bite, right? @tjaychen, just trying to understand if you have any specific concerns that we should be aware of.

Reply:
ah i should probably clarify.  This depends on a lot of other elements in the ROM.

For example, if we have poll loops in the rom on analog components, and those components never respond for whatever reason, it would guarantee that a watchdog would always trigger in ROM and put you into a state where you might be able to get a lot of debug info. 

So putting the value to be long enough such that later code can address it would definitely be good, but unfortunately for ROM we'd have to think about if there are any corner conditions that might get the code stuck in places and the watchdog effect would be bad.

Reply:
I've changed it so that in the TEST and RMA states the watchdog timer will now get disabled and in the production and dev states it will be enabled and the threshold set to a value read from OTP (which could be set to the max, so ~6 hours).

Reply:
sounds good, thanks @mundaym